src/test/scala/mesosphere/marathon/core/storage/store/impl/cache/LazyCachingPersistenceStoreTest.scala

```diff
@@ -31,9 +36,139 @@ class LazyCachingPersistenceStoreTest extends AkkaUnitTest
     new ZkPersistenceStore(client, Duration.Inf, 8)
   }
 
-  private def cachedZk = new LazyCachingPersistenceStore(zkStore)
+  private def cachedZk = LazyCachingPersistenceStore(zkStore)
 
   behave like basicPersistenceStore("LazyCache(InMemory)", cachedInMemory)
   behave like basicPersistenceStore("LazyCache(Zk)", cachedZk)
   // TODO: Mock out the backing store.
+
+  behave like cachingPersistenceStore("cache internals(InMemory)", cachedInMemory)
+
+  def cachingPersistenceStore[K, C, Serialized](
+    name: String,
+    newStore: => LazyCachingPersistenceStore[K, C, Serialized])(
+    implicit
+    ir: IdResolver[String, TestClass1, C, K],
+    m: Marshaller[TestClass1, Serialized],
+    um: Unmarshaller[Serialized, TestClass1]): Unit = {
+
+    name should {
+      "purge the cache appropriately" in {
+        implicit val clock = new SettableClock()
+        val store = newStore
+        for (i <- 1 to 100) {
```

Comment:
no for loops in storage at all, In general please try to remove them from your scala practices. Please use `1.to(100).foreach()`